packages/turbo-workspace-convert/src/utils.ts

```diff
@@ -0,0 +1,239 @@
+  const updated: Array<string> = [];
+  project.workspaceData.workspaces.forEach((workspace) => {
+    const { name } = workspace;
+    if (dependencyList[name]) {
+      dependencyList[name] = to.name === "pnpm" ? `workspace:*` : "*";
+      updated.push(name);
+    }
+  });
+
+  return { dependencyList, updated };
+}
+
+function updateDependencies({
+  project,
+  workspace,
+  to,
+  logger,
+  options,
+}: {
+  workspace: Workspace;
+  project: Project;
+  to: PackageManagerDetails;
+  logger: Logger;
+  options?: Options;
+}) {
+  const workspacePackageJson = getPackageJson({
+    workspaceRoot: workspace.paths.root,
+  });
+
+  // update all dependencies
+  const stats: Record<string, Array<string>> = {
+    dependencies: [],
+    devDependencies: [],
+  };
+  const { devDependencies, dependencies } = workspacePackageJson;
+  if (devDependencies) {
+    const { updated, dependencyList } = updateDependencyList({
+      dependencyList: devDependencies,
+      project,
+      to,
+    });
+    workspacePackageJson["devDependencies"] = dependencyList;
+    stats.devDependencies = updated;
+  }
+  if (dependencies) {
+    const { updated, dependencyList } = updateDependencyList({
+      dependencyList: dependencies,
+      project,
+      to,
+    });
+    workspacePackageJson["dependencies"] = dependencyList;
+    stats.dependencies = updated;
+  }
+
+  const numDevDependencies = stats.devDependencies.length;
+  const numDependencies = stats.dependencies.length;
+  const suffix = (num: number) => `${num <= 1 ? "y" : "ies"}`;
+
+  // both updated
+  if (numDependencies > 0 && numDevDependencies > 0) {
+    logger.workspaceStep(
+      `updating ${chalk.green(
+        numDevDependencies.toString()
+      )} devDependenc${suffix(numDependencies)} and ${chalk.green(
+        numDependencies.toString()
+      )} dependenc${suffix(numDependencies)} in ./${path.relative(
+        project.paths.root,
+        workspace.paths.packageJson
+      )}`
+    );
+  }
+  // only devDependencies updated
+  else if (numDevDependencies > 0) {
+    logger.workspaceStep(
+      `updating ${chalk.green(
+        numDevDependencies.toString()
+      )} devDependenc${suffix(numDevDependencies)} in ./${path.relative(
+        project.paths.root,
+        workspace.paths.packageJson
+      )}`
+    );
+  }
+  // only dependencies updated
+  else if (numDependencies > 0) {
+    logger.workspaceStep(
+      `updating ${chalk.green(numDependencies.toString())} dependenc${suffix(
+        numDependencies
+      )} in ./${path.relative(project.paths.root, workspace.paths.packageJson)}`
+    );
+    // no dependencies updated
+  } else {
+    logger.workspaceStep(
+      `no workspace dependencies found in ./${path.relative(
+        project.paths.root,
+        workspace.paths.packageJson
+      )}`
+    );
+  }
+
+  if (!options?.dry) {
+    fs.writeJSONSync(workspace.paths.packageJson, workspacePackageJson, {
+      spaces: 2,
+    });
+  }
```

Comment:
This doesn't feel very much like a "utility". There is application logic tucked in here.